examples/sfu-ws/main.go

```diff
@@ -0,0 +1,216 @@
+	m := webrtc.MediaEngine{}
+
+	// Setup the codecs you want to use.
+	m.RegisterCodec(webrtc.NewRTPVP8Codec(webrtc.DefaultPayloadTypeVP8, 90000))
+	m.RegisterCodec(webrtc.NewRTPOpusCodec(webrtc.DefaultPayloadTypeOpus, 48000))
+
+	// Create the API object with the MediaEngine
+	api = webrtc.NewAPI(webrtc.WithMediaEngine(m))
+
+}
+
+func Room(w http.ResponseWriter, r *http.Request) {
+
+	// Websocket client
+	c, err := upgrader.Upgrade(w, r, nil)
+	checkError(err)
+	defer c.Close()
+
+	// Read sdp from websocket
+	mt, msg, err := c.ReadMessage()
+
+	if !hasPublisher {
+		fmt.Println("publisher sdp:=============\n", string(msg))
+		hasPublisher = true
+
+		// Create a new RTCPeerConnection
+		PubReceiver, err = api.NewPeerConnection(peerConnectionConfig)
+		checkError(err)
+
+		PubReceiver.OnTrack(func(remoteTrack *webrtc.Track, receiver *webrtc.RTPReceiver) {
+			if remoteTrack.PayloadType() == webrtc.DefaultPayloadTypeVP8 || remoteTrack.PayloadType() == webrtc.DefaultPayloadTypeVP9 || remoteTrack.PayloadType() == webrtc.DefaultPayloadTypeH264 {
+
+				// Create a local video track, all our SFU clients will be fed via this track
+				var err error
+				videoTrack, err = PubReceiver.NewTrack(remoteTrack.PayloadType(), remoteTrack.SSRC(), "video", "pion")
+				checkError(err)
+
+				// Send a PLI on an interval so that the publisher is pushing a keyframe every rtcpPLIInterval
+				go func() {
+					ticker := time.NewTicker(rtcpPLIInterval)
+					for range ticker.C {
+						if err := PubReceiver.SendRTCP(&rtcp.PictureLossIndication{MediaSSRC: videoTrack.SSRC()}); err != nil {
+							fmt.Println(err)
+						}
+					}
+				}()
+
+				rtpBuf := make([]byte, 1400)
+				for {
+					i, err := remoteTrack.Read(rtpBuf)
+					checkError(err)
+					_, err = videoTrack.Write(rtpBuf[:i])
+					checkError(err)
+				}
+
+			} else {
+
+				// Create a local audio track, all our SFU clients will be fed via this track
+				var err error
+				audioTrack, err = PubReceiver.NewTrack(remoteTrack.PayloadType(), remoteTrack.SSRC(), "audio", "pion")
+				checkError(err)
+
+				rtpBuf := make([]byte, 1400)
+				for {
+					i, err := remoteTrack.Read(rtpBuf)
+					checkError(err)
+					_, err = audioTrack.Write(rtpBuf[:i])
+					checkError(err)
+				}
+			}
+		})
+
+		offer := webrtc.SessionDescription{
+			SDP:  string(msg),
+			Type: webrtc.SDPTypeOffer,
+		}
+
+		// Set the remote SessionDescription
+		checkError(PubReceiver.SetRemoteDescription(offer))
+
+		// Create answer
+		answer, err := PubReceiver.CreateAnswer(nil)
+		checkError(err)
+
+		// Sets the LocalDescription, and starts our UDP listeners
+		checkError(PubReceiver.SetLocalDescription(answer))
+
+		// Send server sdp to publisher
+		checkError(c.WriteMessage(mt, []byte(answer.SDP)))
+		fmt.Println("send pion sdp to publisher:==============\n" + answer.SDP)
+	} else {
+		fmt.Println("scriber sdp:=============\n", string(msg))
+
+		recvOnlyOffer := webrtc.SessionDescription{
```

Comment:
Can you inline move this variable by where it is used?

Reply:
Yes, i'll fix